src/aleph/chains/common.py

```diff
@@ -139,16 +119,23 @@ async def incoming(
     item_hash = pending_message.item_hash
     sender = pending_message.sender
     confirmations = []
+    chain_name = tx_context.chain if tx_context is not None else None
     ids_key = (item_hash, sender, chain_name)
 
-    if chain_name and tx_hash and height:
+    if tx_context is not None:
         if seen_ids is not None:
             if ids_key in seen_ids.keys():
-                if height > seen_ids[ids_key]:
+                if tx_context.height > seen_ids[ids_key]:
                     return IncomingStatus.MESSAGE_HANDLED, []
 
         confirmations.append(
-            MessageConfirmation(chain=chain_name, hash=tx_hash, height=height)
+            MessageConfirmation(
```

Comment:
```python
@dataclass
class TxContext:
    chain: str
    hash: str
    height: int
    # Transaction timestamp, in Unix time (number of seconds since epoch).
    time: int
    publisher: str
```

```
class MessageConfirmation(BaseModel):
    chain: Chain
    height: int
    hash: str
    time: float
    publisher: str
```

Reply:
I redefined `MessageConfirmation` because the fields are not yet available in aleph-message. It still makes sense to split `TxContext` and  `MessageConfirmation` because we might add fields later on to  `TxContext` that we do not necessarily want to expose. Open to suggestions here, I could also use only one class. One thing I'll do is set `TxContext.time` to float, regardless.

Reply:
What about using inheritance between them ?
Something like:
```python
class TxContext(MessageConfirmation):
    """Transaction context from chain. Add attributes for additional context."""
    pass
```